### DISCUSSION TITLE
Failed to resolve assembly: netstandard, Version=2.0.0.0

### DISCUSSION THREAD

**Discussion by MarshMello0** [jbevain/cecil#802] [Q&A]
I've been using Mono.Cecil for my modding software for a while now. I recently added it into the launcher which didn't have it beforehand and for some reason when I release it with `dotnet publish` there is an error with missing netstandard.

> Failed to resolve assembly: 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

If I build it normally, it doesn't have this issue. Only once it's published.
This is the netstandard I have installed in the launcher.
![image](https://user-images.githubusercontent.com/33008390/135433242-d528e5b4-ea8f-4758-a328-55c1326815f5.png)

Here are some extra details which may help.
It's a WPF application targeting Windows x64 only
Its made in .net 5.0
I'm modifying some a games code made in Unity.

This is the method where it's having the error
https://gitlab.com/vtolvr-mods/ModLoader/-/blob/dev/Launcher/Views/Settings.xaml.cs#L633

I'm unsure what to search for this issue or how I should debug it, so any help would be greatly appreciated.
Thank you.

**Comment by jbevain** [comment]
Hi,

Unfortunately it's hard to say what's happening, I think you're going to need to debug what's going, and find out what requires the reference to the netstandard assembly and why it can't be resolved.

**Reply by Nezz** [reply]
I have the same issue. It happens for me under the following circumstances:
- Declare a class that inherits from `Newtonsoft.Json.JsonConverter` in a .NET Standard 2.1 library project
- Include a version Newtonsoft.Json that is compiled against .NET Standard 2.0. Right now they don't ship any DLLs compiled against .NET Standard 2.1.
- Recursively resolve the base types

The end result is this:
```
Failed to resolve System.Object for Newtonsoft.Json.JsonConverter:
Failed to resolve assembly: 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
```

The search directory contains `netstandard.dll`, but it's version 2.1.0.0.
